service/share/cache_availability.go

```diff
@@ -0,0 +1,64 @@
+package share
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/autobatch"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var (
+	// DefaultWriteBatchSize defines the size of the batched header write.
+	// Headers are written in batches not to thrash the underlying Datastore with writes.
+	// TODO(@Wondertan, @renaynay): Those values must be configurable and proper defaults should be set for specific node
+	//  type. (#709)
+	DefaultWriteBatchSize = 2048
+
+	cacheAvailabilityPrefix = datastore.NewKey("sampling_result")
+)
+
+func rootKey(root *Root) datastore.Key {
+	return datastore.NewKey(root.String())
+}
+
+// cacheAvailability wraps a given Availability (whether it's light or full)
+// and stores the results of a successful sampling routine over a given Root's hash
+// to disk.
+type cacheAvailability struct {
```

Comment:
I wonder if sth like `PersistedAvailability` could have been a better name?

Reply:
Yeah could be.